src/components/Toast/Toast.tsx

```diff
@@ -1,91 +1,75 @@
    */
-  dismissible?: boolean;
+  variant: Variant;
   /**
-   * Name of toast SVG icon. Available options are: toast, error, warning-2, success
+   * Callback when Toast is dismissed.
    */
-  iconName?: string;
-  /**
-   * The title attribute for the SVG icon in the toast
-   */
-  iconTitle?: string;
-  /**
-   * Stylistic variations for the toast type.
-   * - **success** - results in a green toast
-   * - **warning** - results in a yellow toast
-   * - **error** - results in a red toast
-   * - **info** - results in a blue toast
-   */
-  variant?: 'success' | 'warning' | 'error' | 'info' | 'brand';
-}
+  onDismiss?: () => void;
+};
 
 /**
- * Primary UI component for user interaction
+ * ```ts
+ * import {Toast} from "@chanzuckerberg/eds-components";
+ * ```
+ *
+ * A toast used to provide information on the state of the page, usually in response to a
+ * user action. Ex: The user updates their profile, and a toast pop-up informs them that the
+ * data was successfully saved.
  */
-export const Toast: React.FC<Props> = ({
-  iconTitle,
+export const Toast = ({
+  text,
   className,
-  children,
-  iconName,
-  dismissible = true,
   variant,
-  closeButtonText = 'Close',
+  onDismiss,
+  // Allow for additional attributes such as aria roles
   ...other
-}) => {
-  const componentClassName = clsx(styles['toast'], className, {
-    [styles['toast--success']]: variant === 'success',
-    [styles['toast--warning']]: variant === 'warning',
-    [styles['toast--error']]: variant === 'error',
-    [styles['toast--brand']]: variant === 'brand',
-  });
-  const [dismissed, setDismissed] = useState(false);
-
-  function onDismiss(e: any) {
-    e.preventDefault();
-    setDismissed(true);
-  }
-  return dismissed ? null : (
-    <div className={componentClassName} role="alert" {...other}>
-      {iconName && (
-        <Icon
-          name={iconName}
-          title={iconTitle}
-          className={styles['toast__icon']}
-        />
-      )}
-      <div className={styles['toast__body']}>{children}</div>
-
-      {dismissible && (
-        <Button
-          className={styles['toast__close-btn']}
-          variant="bare"
-          aria-label={closeButtonText}
-          iconName="x"
-          iconPosition="after"
-          onClick={(e: any) => onDismiss(e)}
-        />
+}: Props) => {
+  return (
+    <div
+      className={clsx(
+        className,
+        styles.toast,
+        /* TODO: uncomment and point to corresponding style classes once Notifications is fully migrated */
+        // variant === 'success' && colorStyles.colorSuccess,
+        // variant === 'alert' && colorStyles.colorAlert,
       )}
+      {...other}
+    >
+      <div className={styles['toast--content']}>
+        {/* TODO: point to internal Text and NotificaitonIcon once fully migrated */}
+        {/* <NotificationIcon variant={variant} />
+        <Text color="inherit" size="sm">
+          {text}
+        </Text> */}
```

Comment:
Can this use a typography mixin instead of going through the Text component?

Reply:
Will use mixin